src/package_manager/apt.ts

```diff
@@ -62,6 +62,33 @@ const distributionSpecificAptDependencies = {
 		// libc++-dev and libc++abi-dev installs intentionally removed because:
 		// https://github.com/ros-tooling/setup-ros/issues/506
 	],
+	noble: [
+		// Basic development packages (from ROS 2 source/development setup instructions)
+		// ros-dev-tools includes many packages that we needed to include manually in Focal & older
+		"python3-flake8-docstrings",
+		"python3-pip",
+		"python3-pytest-cov",
+		"python3-flake8-blind-except",
+		"python3-flake8-builtins",
+		"python3-flake8-class-newline",
+		"python3-flake8-comprehensions",
+		"python3-flake8-deprecated",
+		"python3-flake8-import-order",
+		"python3-flake8-quotes",
+		"python3-pytest-repeat",
+		"python3-pytest-rerunfailures",
+		"ros-dev-tools",
+		// Additional colcon packages (not included in ros-dev-tools)
+		"python3-colcon-coveragepy-result",
+		"python3-colcon-lcov-result",
+		"python3-colcon-meson",
+		"python3-colcon-mixin",
+		// FastRTPS dependencies
+		"libasio-dev",
+		"libtinyxml2-dev",
```

Comment:
Again, if we are using `rosdep`, this shouldn't be necessary because it is already depended on by https://github.com/ros2/tinyxml2_vendor/blob/rolling/package.xml